controllers/capabilities/openapi_product_reconciler.go

```diff
@@ -369,3 +376,125 @@ func (p *OpenAPIProductReconciler) desiredPrivateAPISecurity() *capabilitiesv1be
 
 	return privateAPISec
 }
+
+func (p *OpenAPIProductReconciler) desiredOIDCAuthentication(secReq *helper.ExtendedSecurityRequirement) *capabilitiesv1beta1.AuthenticationSpec {
+
+	if p.openapiCR.Spec.OIDC == nil {
+		return nil
+	}
+
+	authSpec := &capabilitiesv1beta1.AuthenticationSpec{
+		OIDC: &capabilitiesv1beta1.OIDCSpec{
+			IssuerType: p.openapiCR.Spec.OIDC.IssuerType,
+			Security:   p.desiredPrivateAPISecurity(),
+			AuthenticationFlow: &capabilitiesv1beta1.OIDCAuthenticationFlowSpec{
+				StandardFlowEnabled:       false,
+				ImplicitFlowEnabled:       false,
+				DirectAccessGrantsEnabled: false,
+				ServiceAccountsEnabled:    false,
+			},
+		},
+	}
+
+	if secReq.Value.Type == "openIdConnect" {
+		p.setOIDCAuthenticationParams(authSpec, secReq)
+	} else { // oauth2
+		p.setOauth2AuthenticationParams(authSpec, secReq)
+	}
+
+	return authSpec
+}
+
+func (p *OpenAPIProductReconciler) setOIDCAuthenticationParams(authSpec *capabilitiesv1beta1.AuthenticationSpec, secReq *helper.ExtendedSecurityRequirement) {
+	tmpHeaders := "headers"
+	authSpec.OIDC.CredentialsLoc = &tmpHeaders
+
+	issuerEndpoint, err := p.getIssuerEndpoint()
+	if err != nil {
+		p.Logger().Info("can't get issuerEndpoint value from OpenAPI CR; not set\n")
+		issuerEndpoint = ""
+	}
+
+	if p.openapiCR.Spec.OIDC != nil {
+		authSpec.OIDC.IssuerType = p.openapiCR.Spec.OIDC.IssuerType
+		authSpec.OIDC.IssuerEndpoint = issuerEndpoint
```

Comment:
what about `authSpec.OIDC.IssuerEndpoint =  p.openapiCR.Spec.OIDC,IssuerEndpoint` ?

Reply:
done. Thank you!